.github/workflows/pdk-basic.yml

```diff
@@ -0,0 +1,52 @@
+# This is a basic workflow to help you get started with Actions
+
+name: PDK basic
+
+# Controls when the workflow will run
+on:
+  workflow_call:
+    inputs:
+      container_image:
+        description: Image to use when running tests
+        default: 'puppet/puppet-dev-tools:2022-03-28-92c7176'
+        required: false
+        type: string
+
+
+env:
+  GITHUB_USER: ${{ secrets.PUPPET_REPO_USER }}
+  GITHUB_TOKEN: ${{ secrets.PUPPET_REPO_USER_TOKEN }}
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  validate:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        puppet-version: [5, 6, 7]
+    container: ${{ inputs.container_image }}
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      - name: action-pdk-validate-puppet-${{ matrix.puppet-version }}
+        run: pdk validate --puppet-version=${{ matrix.puppet-version }}
+      # Runs pdk test unit
+
+  unit-puppet:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        puppet-version: [5, 6, 7]
```

Comment:
We have https://github.com/voxpupuli/gha-puppet/blob/v1/.github/workflows/basic.yml#L55-L57 that gets all major puppet versions from the metadata.json and build the matrix dynamically. I'm wondering if this could be ported to here and if it would be worth the effort.

Reply:
If you're going to specify it somewhere, why not encourage people to do it in `metadata.json` rather than in `workflow.yaml`?

Reply:
I am going to update the workflow to use the suggested method.  My only concern is that metadata doesn't really support future versions of puppet.  So if a modules is only compatible with puppet 6,7 how do we test for 8 before the metadata is updated to 8?

I also filed: https://github.com/puppetlabs/puppet-dev-tools/pull/113  to add puppet-metadata

Reply:
> So if a modules is only compatible with puppet 6,7 how do we test for 8 before the metadata is updated to 8?

It should handle that well since it iterates the major versions range:
https://github.com/voxpupuli/puppet_metadata/blob/bda0b0f62782492f2f8d5d3392e8aa6d02faff72/lib/puppet_metadata/metadata.rb#L138-L151

As you can see there, it doesn't handle the unbounded dependency automatically. `>= 7.0` will only be tested on 7 until we change the hardcoded upper limit. I didn't know how to solve that otherwise. In practice I also think it's uncommon.

Reply:
Also how do I handle the case when the user does not have metadata or does not have the puppet version metadata.  I don't know if this would occur.  Is there a default value that can be supplied or friendly error to update the metadata?